specs/BrowserTaskManager.md

```diff
@@ -0,0 +1,81 @@
+# Background
+The WebView2 team has been asked to provide an API to expose the Browser Task
+Manager. The Browser Task Manager is a window helpful for debugging that 
+displays the different processes associated with the current browser process and 
+what they're used for. For instance, it could denote that a particular process 
+is a renderer process and would show the different web pages rendered by that 
+process. In a chromium browser it can be opened by the end user by pressing 
+`Shift+Esc` or from the DevTools window's title bar's context menu entry 
+`Browser task manager`. For WebView2 applications, we block the 
+`Shift+Esc` shortcut, but the task manager can still be opened via the 
+`Browser task manager` entry.
+
+At the current time, it is expected that the user will close the Task 
+Manager manually, so we do not need to provide an API to close it.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` to provide an `OpenTaskManagerWindow` 
+method. This method will open a new window containing the task manager. 
+If the task manager is already opened, this method will do nothing.
+
+# Examples
+## C++: Open Task Manager
+
+``` cpp
+wil::com_ptr<ICoreWebView2_5> m_webview;
+
+// This method could be called from a menu bar item, such as 
+// [Script -> Open Task Manager Window]. 
+void ScriptComponent::OpenTaskManagerWindow()
+{
+    CHECK_FAILURE(m_webview->OpenTaskManagerWindow());
+}
+```
+
+## C#: Open Task Manager
+```c#
+private WebView2 m_webview;
+
+// This method could be called from a menu bar item, such as 
+// [Script -> Open Task Manager Window]. 
+void OpenTaskManagerWindow()
+{
+    m_webview.CoreWebView2.OpenTaskManagerWindow();
+}
+```
+
+# API Details
+## C++
+```
+/// This is a continuation of the `ICoreWebView2_4` interface
+[uuid(20d02d59-6df2-42dc-bd06-f98a694b1302), object, pointer_default(unique)]
+interface ICoreWebView2_5 : ICoreWebView2_4 {
+    /// Opens the Browser Task Manager view as a new window. Does nothing
+    /// if run when the Browser Task Manager is already open.
```

Comment:
Does the Browser Task Manager receive foreground as a result of this call? Does it depend on whether the Browser Task Manager is already open?

Reply:
Good question! We should ensure that the window is given foreground if the window was already open or not (might be best effort if we don't have foreground permission). And update the docs to note this.

Reply:
Yes, the Browser Task Manager will receive foreground in both cases - when it is opened as a new window or when it is already open. I have updated the doc to reflect this.